C++/test/algorithm/sorting/sorting.cpp

```diff
@@ -0,0 +1,82 @@
+#include <random>
+#include <limits>
+#include "third_party/catch.hpp"
+#include "algorithm/sorting/bubble_sort.hpp"
+
+// Prototypes
+int generate_random_int(int, int);
+vector<int> generate_unsorted_vector();
+
+// Pointer to function
+using sorting_function = void(*)(vector<int>&, int, bool);
+
+TEST_CASE("Sort in ascending order", "[sorting]") {
+	// Sorting algorithms
+	vector<sorting_function> sorting_functions = {
+			bubble_sort
+	};
+
+	vector<int> original, algo_sorted, std_sorted;
+
+	int times_to_run = 20;
+	while (times_to_run--) {
+		original = algo_sorted = std_sorted = generate_unsorted_vector();
+		std::sort(std_sorted.begin(), std_sorted.end());
+
+		// Run tests
+		for (auto sorting_algo : sorting_functions) {
+			sorting_algo(algo_sorted, 1, false);
+			REQUIRE(algo_sorted == std_sorted);
+			algo_sorted = original;
+		}
+	}
+}
+
+TEST_CASE("Sort in descending order", "[sorting]") {
+	// Sorting algorithms
+	vector<sorting_function> sorting_functions = {
+			bubble_sort
+	};
+
+	vector<int> original, algo_sorted, std_sorted;
+
+	int times_to_run = 20;
+	while (times_to_run--) {
+		original = algo_sorted = std_sorted = generate_unsorted_vector();
+		std::sort(std_sorted.rbegin(), std_sorted.rend());
+
+		// Run tests
+		for (auto sorting_algo : sorting_functions) {
+			sorting_algo(algo_sorted, -1, false);
+			REQUIRE(algo_sorted == std_sorted);
+			algo_sorted = original;
+		}
+	}
+}
+
+/*
+	generate_unsorted_vector
+	------------------------
+	Generates a vector with a random size (up to 100) and fills with random integers.
```

Comment:
Based on the code, the comment should say `up to 1000` (instead of 100). Also, I think the following will be better:

```
Creates a vector of random size (up to 1000) and populates it with random integers.
```

It would also be better if the function accepted an argument `max_size` and had 1000 as the default value for it if none was provided. This would make it more flexible, and therefore reusable in even more test case scenarios.